Sync changes with os_linux_sparc.cpp
Support setting thread stack sizes on sparc64

Index: hotspot/src/os_cpu/bsd_sparc/vm/os_bsd_sparc.cpp
--- hotspot/src/os_cpu/bsd_sparc/vm/os_bsd_sparc.cpp.orig
+++ hotspot/src/os_cpu/bsd_sparc/vm/os_bsd_sparc.cpp
@@ -618,7 +618,7 @@ JVM_handle_bsd_signal(int sig,
       return 1;
     }
 
-    if (checkPrefetch(uc, pc)) {
+    if ((sig == SIGSEGV || sig == SIGBUS) && checkPrefetch(uc, pc)) {
       return 1;
     }
 
@@ -740,7 +740,7 @@ bool os::is_allocatable(size_t bytes) {
 
 size_t os::Bsd::min_stack_allowed  = 128 * K;
 
-bool os::Bsd::supports_variable_stack_size() {  return false; }
+bool os::Bsd::supports_variable_stack_size() {  return true; }
 
 // return default stack size for thr_type
 size_t os::Bsd::default_stack_size(os::ThreadType thr_type) {
